std/algorithm/mutation.d

```diff
@@ -3128,6 +3139,25 @@ if (isBlitAssignable!T && !is(typeof(lhs.proxySwap(rhs))))
     swap(a3, a4);
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=21429
+@safe unittest
+{
+    enum e = (){
+        Tuple!int a = 5, b = 6;
+        swap(a, b);
+        assert(a[0] == 6);
+        assert(b[0] == 5);
+        return 0;
+    }();
+}
+
+/// ditto
+void swap(T)(ref T lhs, ref T rhs)
+if (is(typeof(lhs.proxySwap(rhs))))
+{
+    lhs.proxySwap(rhs);
+}
```

Comment:
@ntrel
What was this overload supposed to solve?

Reply:
Unfortunately, I haven’t been able to figure that out myself.
All I can tell is that it breaks this code:
https://github.com/dlang/phobos/blob/336bed6d8ffec74d117b755866c5bd22e3d610a1/std/algorithm/sorting.d#L2145